pyccl/tests/test_cclobject.py

```diff
@@ -33,111 +29,20 @@ def test_fancy_repr():
-    assert 2*PK1 != PK2
-
-    # 3.1. Using a factorizable Tk3D object.
-    a_arr, lk_arr, pk_arr = PK1.get_spline_arrays()
-    TK1 = ccl.Tk3D(a_arr=a_arr, lk_arr=lk_arr,
-                   pk1_arr=pk_arr, pk2_arr=pk_arr, is_logt=False)
-    TK2 = ccl.Tk3D(a_arr=a_arr, lk_arr=lk_arr,
-                   pk1_arr=pk_arr, pk2_arr=pk_arr, is_logt=False)
-    TK3 = ccl.Tk3D(a_arr=a_arr, lk_arr=lk_arr,
-                   pk1_arr=2*pk_arr, pk2_arr=2*pk_arr, is_logt=False)
-    assert TK1 == TK2
-    assert TK1 != TK3
-
-    # 3.2. Using a non-factorizable Tk3D object.
-    a_arr_2 = np.arange(0.5, 0.9, 0.1)
-    lk_arr_2 = np.linspace(-2, 1, 8)
-    TK1 = ccl.Tk3D(
-        a_arr=a_arr_2, lk_arr=lk_arr_2,
-        tkk_arr=np.ones((a_arr_2.size, lk_arr_2.size, lk_arr_2.size)))
-    TK2 = ccl.Tk3D(
-        a_arr=a_arr_2, lk_arr=lk_arr_2,
-        tkk_arr=np.ones((a_arr_2.size, lk_arr_2.size, lk_arr_2.size)))
-    assert TK1 == TK2
-
-    # 4. Using a CosmologyCalculator.
-    pk_linear = {"a": a_arr,
-                 "k": np.exp(lk_arr),
-                 "delta_matter:delta_matter": pk_arr}
-    COSMO1 = ccl.CosmologyCalculator(
-        Omega_c=0.25, Omega_b=0.05, h=0.67, n_s=0.96, sigma8=0.81,
-        pk_linear=pk_linear, pk_nonlin=pk_linear)
-    COSMO2 = ccl.CosmologyCalculator(
-        Omega_c=0.25, Omega_b=0.05, h=0.67, n_s=0.96, sigma8=0.81,
-        pk_linear=pk_linear, pk_nonlin=pk_linear)
-    assert COSMO1 == COSMO2
-
-    # 5. Using a Tracer object.
-    TR1 = ccl.CMBLensingTracer(cosmo, z_source=1101)
-    TR2 = ccl.CMBLensingTracer(cosmo, z_source=1101)
-    assert TR1 == TR2
-
-
-def test_CCLHalosObject():
-    # Test eq --> repr <-- hash for all kinds of CCL halo objects.
-
-    # 1. Build a halo model calculator using the default parametrizations.
-    cosmo = ccl.CosmologyVanillaLCDM(transfer_function="bbks")
-    HMC = ccl.halos.HMCalculator(
-        cosmo, massfunc="Tinker08", hbias="Tinker10", mass_def="200m")
-
-    # 2. Define separate default halo model ingredients.
-    MDEF = ccl.halos.MassDef200m()
-    HMF = ccl.halos.MassFuncTinker08(cosmo, mass_def=MDEF)
-    HBF = ccl.halos.HaloBiasTinker10(cosmo, mass_def=MDEF)
-
-    # 3. Test equivalence.
-    assert MDEF == HMC._mdef
-    assert HMF == HMC._massfunc
-    assert HBF == HMC._hbias
-    HMC2 = ccl.halos.HMCalculator(
-        cosmo, massfunc=HMF, hbias=HBF, mass_def=MDEF)
-    assert HMC == HMC2
-
-    # 4. Test halo profiles.
-    CM1 = ccl.halos.Concentration.from_name("Duffy08")()
-    CM2 = ccl.halos.ConcentrationDuffy08()
-    assert CM1 == CM2
-
-    # TODO: uncomment once __eq__ methods are implemented.
-    # P1 = ccl.halos.HaloProfileHOD(c_M_relation=CM1)
-    # P2 = ccl.halos.HaloProfileHOD(c_M_relation=CM2)
-    # assert P1 == P2
-
-    # PCOV1 = ccl.halos.Profile2pt(r_corr=1.5)
-    # PCOV2 = ccl.halos.Profile2pt(r_corr=1.0)
-    # assert PCOV1 != PCOV2
-    # PCOV2.update_parameters(r_corr=1.5)
-    # assert PCOV1 == PCOV2
```

Comment:
There was no point of this test living in here so I distributed it to the respective test modules.